tmt/result.py

```diff
@@ -55,7 +55,18 @@ def is_result_outcome(cls, value: 'ResultInterpret') -> bool:
     }
 
 
-ResultData = Dict[str, Any]
+@dataclasses.dataclass
+class ResultData():
+    """
+    Formal data class containing information about result of the test
+
+    Replacement of Dict[str, Any]
+    """
+    result: ResultOutcome
+    log: Optional[List[str]] = None
```

Comment:
Is `log = None` better in some way than `log = []`? Emptiness can be tested in both cases, `not data.log` would work with an empty list nicely, and we wouldn't have to worry about `Optional`. I'd go for `default_factory=list`, reducing the complexity for users of `log`.

Reply:
Solved as discussed: `log: List[str] = dataclasses.field(default_factory=list)`